src/main/java/gr/cytech/chatreminderbot/rest/controlCases/CaseSetReminder.java

```diff
@@ -105,12 +120,41 @@ String buildAndPersistReminder(Request request) {
         }
 
         timerSessionBean.setTimerForReminder(reminder);
+        Map<String, String> parameters = new LinkedHashMap<>();
+        parameters.put("reminderId", String.valueOf(reminder.getReminderId()));
+        parameters.put("name", reminder.getSenderDisplayName());
+
+        if (request.getAction() != null) {
+            logger.info("Button Clicked Update the card message");
+            return buildReminderResponse(reminder, timeToNotify, parameters,
+                    "UPDATE_MESSAGE", request.getAction().getActionMethodName());
+        }
+        logger.info("returned default new message for reminder");
+        return buildReminderResponse(reminder, timeToNotify, parameters,
+                "NEW_MESSAGE", "");
 
-        return "Reminder with text:\n <b>" + reminder.getWhat() + "</b>.\n"
-                + "Saved successfully and will notify you in: \n<b>"
-                + timeToNotify + "</b>";
     }
 
+    private String buildReminderResponse(Reminder reminder, String timeToNotify, Map<String,
+            String> parameters, String typeForMessage, String actionName) {
+        if (actionName.equals(REMIND_AGAIN_IN_10_MINUTES) || actionName.equals(REMIND_AGAIN_NEXT_WEEK)
```

Comment:
this if contains almost the exact same code with its respective implicit else... and contains an awful lot of hard-coded strings